packages/api/src/module/read/course-progress/course-progress.read-module.ts

```diff
@@ -25,4 +27,35 @@ export class CourseProgressReadModule {
       },
     });
   }
+
+  @OnEvent('LearningMaterialsTasks.TaskWasUncompleted')
+  async onTaskWasUncompleted(event: ApplicationEvent<TaskWasUncompleted>) {
+    const where = { learningMaterialsId: event.data.learningMaterialsId };
+    const courseProgress = await this.prismaService.courseProgress.findUnique({ where });
+
+    if (!courseProgress || courseProgress.learningMaterialsCompletedTasks === 0) {
```

Comment:
It's interesting. Very nice that you have spotted such edge case! :) 
We will think more carefully about such situations later (why something like that happened, if yes, what we should do? Such event probably should never come here? What about events ordering?), leave it as you have done for now. I'm working about handling logic right now, so maybe in the future we will change something :) 

Very good job!

Reply:
I think this check is unnecessary, we can trust TaskWasUncompleted (is only for previously Completed task), because it's our internal event. 
And the check may also return false positives, because it's async - as I've stated in previous comment.